src/android/com/megster/cordova/BluetoothSerialService.java

```diff
@@ -296,25 +195,24 @@ public synchronized void connected(BluetoothSocket socket, BluetoothDevice devic
             mInsecureAcceptThread = null;
         }
 
+        mConnectedThread = new ConnectedThread(socket, socketType, device);
+        // Starting the connect thread a second later, as for SPP devices with stored data, they start transmitting as soon as they connect.
+        // If we have not subscribed to listed to them till that time, the data might get lost. Thus we want the connect event message to be passed from plugin to PCMT and let PCMT subscribe for data during this time
+        Handler handler=new Handler(Looper.getMainLooper());
```

Comment:
Starting the connect thread a second later, as for SPP devices with stored data, they start transmitting as soon as they connect. If we have not subscribed to listen to them till that time, the data might get lost. Thus we want the connect event message to be passed from plugin to PCMT and let PCMT subscribe for data during this time